bpf/process/types/basic.h

```diff
@@ -2631,26 +2627,26 @@ read_call_arg(void *ctx, struct msg_generic_kprobe *e, int index, int type,
 	case size_type:
 	case s64_ty:
 	case u64_ty:
-		probe_read(args, sizeof(__u64), &arg);
+		*(__u64 *)args = arg;
```

Comment:
This does not seem safe to me. We do not do nothing to ensure that the u64 values are properly aligned in the args array, and accessing unaligned values can cause issues. I think for x86_64 is fine, but I'm not sure about ARM. To be on the safe side, I'd rather we just use `probe_read` as we did before.

Reply:
Unaligned is safe on ARM. If it isn't then this should fail tests on ARM. The changes I made are more expressive of what arg actually refers to, and avoids a call to a helper.

Reply:
> Unaligned is safe on ARM. If it isn't then this should fail tests on ARM. 

Can you provide a reference for that? The fact that the tests are not failing is not an indication that unaligned access is OK.

Reply:
https://medium.com/@iLevex/the-curious-case-of-unaligned-access-on-arm-5dd0ebe24965 claims support.
Really, we should just add unaligned memory tests. Also, are we trusting probe_read to do the unaligned access, so we should probably consider trusting the compiler to do the same.
I can add unaligned memory tests to the patch.

Reply:
https://stackoverflow.com/questions/38535738/does-aarch64-support-unaligned-access is possibly a more useful reference.

Reply:
Yap, seems to be the case in https://developer.arm.com/documentation/102376/0100/Alignment-and-endianness as well:
> When the address is not a multiple of the element size, the access is unaligned. Unaligned accesses are allowed to addresses marked as Normal, but not to Device regions. An unaligned access to a Device region will trigger an exception (alignment fault).

Reply:
OK, sounds good, let's keep it!